### DISCUSSION TITLE
Float handling

### DISCUSSION THREAD

**Discussion by 25Das** [beholdr/maska#220] [General]
My only problem with maska is how it handles float numbers like 2.5, these numbers are masked like 25 instead of 0.25 or 2.50, either when typed or value assigned to input. Despite that, awesome lib, best I've found and I've being using it since before v1.0.

**Comment by beholdr** [comment]
Did you tried new number mode for that?

**Reply by 25Das** [reply]
3 weeks later, using beta4, I'm having no major problems in my implementation using number mode and normal mask features. Maska format to 2.5 as expected, my perfectionism hungers form '2.50' but I satisfied with being able to consistently use and reuse maska across all my application.
One feedback, I could not make it work with locale as 'br' / 'ptbr' / 'ptBR' / 'pt-BR', so I am using default uk and accepting the space as thousand separator.
Thanks again, awesome lib.

**Reply by beholdr** [reply]
Thank you for kind words.
I don't know Brazilian number format, but with `br` locale it formats number as `1 000,23`. Looks the same with `uk` locale format.

**Reply by 25Das** [reply]
Brazilian number format uses decimal comma and points to separate thousands.

Everything works fine when you input, but when you bind the value using input value property or vue v-model it brokes.

You can test on live demo, set number mode money uk locale to br and `<input class="money-number" value="1234.56">`

The output is: 123.456

With uk locale it masks correctly: 1 234,56

**Reply by beholdr** [reply]
I'll see if I can fix the float conversion for brazilian locale.

**Reply by beholdr** [reply]
Unfortunately, I'm not sure if I can easily fix this issue. The problem is related to the parsing of floating point numbers in the `pt-BR` format. In this format, I have trouble determining whether the `.` character is used to separate thousandths or as a decimal point.

It's easy on a well formed number like `1.234`, but it's not always a case in a mask processing. Such as when the last digit `4` is deleted from a number like `1.234`. In this case, `1.23` should be parsed not as float, but still as an integer.

To solve this issue, you can use a different locale, or you could also try setting the value as `1234,56`: `<input class="money-number" value="1234,56">`
